### PR TITLE
feat: implement get deal by ID functionality with integration tests

### PR DESCRIPTION
**Pull request by YasinduRC**
### Summary

-

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds a get-deal-by-ID endpoint with a service method, mapper mapping, expanded detail DTO, and integration tests. The implementation correctly follows the module's layered conventions: controller delegates to the service interface, the impl uses @Transactional(readOnly=true) with entry/exit logging and ModuleException, the MapStruct mapping is sound, and tests cover happy path, not-found, soft-deleted, and forbidden cases. No must-fix issues found; one minor suggestion about an omitted field.

Found **4** new issue(s):

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java [hunk 1]
```diff
@@ -19,4 +20,10 @@ public class CrmDealDetailResponseDto {
 
 	private CrmOwnerResponseDto owner;
 
+	private CrmDealPriority priority;
+
+	private String contactName;
+
+	private CrmCompanyLookupResponseDto company;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: 🤖 **Claude** · 💬 **Note**

The 'detail' response DTO omits `closingAt`, which exists on the CrmDeal entity and is part of a deal's business data (it's set during createDeal). The happy-path test explicitly asserts `closingAt` does NOT exist, so this appears deliberate — but for a single-deal detail view, the closing date is typically something a client would want. Please confirm the omission is intentional; if not, add `private LocalDateTime closingAt;` (it will map automatically by name).

File: backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java [hunk 2]
```diff
@@ -103,6 +103,7 @@ public interface CrmMapper {
 
 	CrmContactDetailResponseDto crmContactToCrmContactDetailResponseDto(CrmContact contact);
 
+	@Mapping(target = "contactName", source = "contact.name")
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java]: 🤖 **Claude** · 💬 **Note**

The mapper crmDealToCrmDealDetailResponseDto (and the DTO it produces) is shared: it is also used by CrmContactServiceImpl.getContactDetails (CrmContactServiceImpl.java:318) to populate CrmContactDetailResponseDto.deals. Adding `contactName`, `company`, and `priority` to the DTO therefore silently changes the contact-detail endpoint's response (every deal now carries a redundant contactName — the same contact being viewed — plus a company object) and adds N+1 queries: findByContactIdWithAssociations (CrmDealRepositoryImpl.java:267) only fetch-joins `stage` and `owner`, so reading the lazy `contact.name` and `company` now triggers extra selects per deal. Fix: confirm the extra fields are intended in the contact-detail response; if the new fields are only for getDealById, introduce a dedicated DTO for the deal-detail endpoint instead of reusing this one. If reuse is intended, add `contact`/`company` to the fetch joins in findByContactIdWithAssociations to avoid the N+1.

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: 🤖 **Claude** · 💬 **Note**

API shape inconsistency: the list endpoint (CrmDealResponseDto via crmDealToCrmDealResponseDto) exposes the company as a flat `companyName` string and `contactName` string, while this detail DTO exposes a nested `company` object plus a flat `contactName`. Mixing a nested object for company with a flat string for contact within the same DTO, and diverging from the list representation, makes the deal resource's contract inconsistent for clients. Consider aligning the representations (either both nested or both flat) or at least making company/contact symmetric within this DTO.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java [hunk 3]
```diff
@@ -130,6 +130,23 @@ private CrmDealCreateRequestDto validPayload(Long stageId, Long contactId) {
 		return dto;
 	}
 
+	private CrmDeal savedDeal() {
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java:157]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

Test coverage gap: the happy-path test always uses a deal with a company set, so the null-company branch of the detail mapping (CrmDeal.company is nullable per the entity's @JoinColumn) is never exercised for this endpoint. Add a getDealById test for a deal whose contact has no company, asserting the `company` field is null, mirroring the existing createDeal null-company tests.

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java (hunk 1 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: This was intentional because the deal detail view does not require to show the closing date

Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: Updated get contact name and company name via CrmContactLookupResponseDto

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java [hunk 4]
```diff
@@ -264,4 +281,61 @@ void createDeal_ContactWithNoCompany_DealCreatedWithoutCompany() throws Exceptio
...
+			.andExpect(jsonPath(RESULTS_0_PATH + "['id']").value(deal.getId().intValue()))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['name']").value("Saved Deal"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['amount']").value("5000"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['description']").value("Test deal description"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['priority']").value("HIGH"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['contactName']").value("Deal Test Contact"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['company']['name']").value("Deal Company"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['stage']['name']").value("Test Stage"))
+			.andExpect(jsonPath(RESULTS_0_PATH + "['owner']").exists())
+			.andExpect(jsonPath(RESULTS_0_PATH + "['closingAt']").doesNotExist());
```
[the remark this answers is not in the document]
Comment (YasinduRC) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java]: This was intentional because the deal detail view does not require to show the closing dateThis was intentional because the deal detail view does not require to show the closing date

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java (hunk 3 above)
Comment (YasinduRC) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmDealControllerIntegrationTest.java:157]: Address the gap

**Review by sanjula-rtc — DISMISSED** [review]

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java [hunk 5]
```diff
@@ -19,4 +20,10 @@ public class CrmDealDetailResponseDto {
 
 	private CrmOwnerResponseDto owner;
 
+	private CrmDealPriority priority;
+
+	private String contactName;
```
Comment (anusham-rc) [inline backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: we need id also
replace with CrmContactLookupResponseDto

File: backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java [hunk 6]
```diff
@@ -103,6 +103,8 @@ public interface CrmMapper {
...
```
Comment (anusham-rc) [inline backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java]: can remove this mapping

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java [hunk 7]
```diff
@@ -19,4 +20,10 @@ public class CrmDealDetailResponseDto {
...
+
+	private String companyName;
```
Comment (anusham-rc) [inline backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: no need company, check CrmContactLookupResponseDto

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java (hunk 5 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: Fixed

File: backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java (hunk 7 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/payload/response/CrmDealDetailResponseDto.java]: Fixed

File: backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java (hunk 6 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java]: Removed the mapping

File: backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java [hunk 8]
```diff
@@ -103,6 +103,7 @@ public interface CrmMapper {
...
 	CrmDealDetailResponseDto crmDealToCrmDealDetailResponseDto(CrmDeal deal);
```
[the remark this answers is not in the document]
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java]: Created a seperated dto for that and added a criteria query to address the N+1

File: backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java (hunk 2 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/mapper/CrmMapper.java]: Created a separated dto for that and added a criteria query to address the N+1

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 9]
```diff
@@ -281,6 +283,21 @@ public List<CrmDeal> findByContactIdWithAssociations(Long contactId) {
...
 
+	@Override
+	public CrmDeal findByIdWithAssociations(Long id) {
+		CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+		CriteriaQuery<CrmDeal> query = cb.createQuery(CrmDeal.class);
+		Root<CrmDeal> deal = query.from(CrmDeal.class);
+		deal.fetch(CrmDeal_.stage, JoinType.INNER);
+		deal.fetch(CrmDeal_.owner, JoinType.INNER);
+		Fetch<CrmDeal, CrmContact> contactFetch = deal.fetch(CrmDeal_.contact, JoinType.INNER);
+		contactFetch.fetch(CrmContact_.company, JoinType.LEFT);
```
Comment (akilarootcode) [inline backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: why need left join for this

**Review by akilarootcode — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java (hunk 9 above)
Comment (YasinduRC) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: Removed the LEFT JOIN this association is lazily loaded when accessed during mapping, so the API response remains unchanged.

**Review by akilarootcode — APPROVED** [review]

**Review by akilarootcode — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
